lib/iris/tests/unit/cube/test_Cube.py

```diff
@@ -409,6 +409,155 @@ def test_kwargs(self):
         self.assertMaskedArrayEqual(expected_result, res_cube.data)
 
 
+class TestSlices(tests.IrisTest):
+    '''
+    This class tests the capability of iris.cube.Cube.slices(), including its
+    ability to correctly re-order the dimensions.
+    '''
+    def setUp(self):
+        '''
+        setup a 4D iris cube, each dimension is length 1.
+        The dimensions are;
+            dim1: height
+            dim2: time
+            dim3: longitude
+            dim4: latitude
+        '''
+        self.cube = iris.cube.Cube(np.array([[[[8.]]]]))
+        self.cube.add_dim_coord(iris.coords.DimCoord([0], "height"), [0])
+        self.cube.add_dim_coord(iris.coords.DimCoord([0], "time"), [1])
+        self.cube.add_dim_coord(iris.coords.DimCoord([0], "longitude"), [2])
+        self.cube.add_dim_coord(iris.coords.DimCoord([0], "latitude"), [3])
+
+    @staticmethod
+    def expected_cube_setup(dim1name, dim2name, dim3name):
+        '''
+        input:
+        ------
+            dim1name: str
+                name of the first dimension coordinate
+            dim2name: str
+                name of the second dimension coordinate
+            dim3name: str
+                name of the third dimension coordinate
+        output:
+        ------
+            cube: iris cube
+                iris cube with the specified axis holding the data 8
+        '''
+        cube = iris.cube.Cube(np.array([[[8.]]]))
+        cube.add_dim_coord(iris.coords.DimCoord([0], dim1name), [0])
+        cube.add_dim_coord(iris.coords.DimCoord([0], dim2name), [1])
+        cube.add_dim_coord(iris.coords.DimCoord([0], dim3name), [2])
+        return cube
+
+    def make_the_test(self, dim1, dim2, dim3, dim_to_remove):
```

Comment:
As, whenever you use this, you then always say `self.assertEqual(actual, expected)`, I would put that assertion inside it, and rename it e.g. `check_dims_order`.  Each testcase then just calls it with a list of coords.